### PR TITLE
fixing visual studio build

### PR DESCRIPTION
**Pull request by tyler97**
Fixed typo from previous commit.
Added files that weren't being included in visual studio project.
Disabled warning 4146
Changed auto to auto& to avoid extra copies.
Build should work on with Visual Studio now.

### CODE REVIEW

File: src/SourceLine.cpp [hunk 1]
```diff
@@ -3,11 +3,11 @@
 #include "SourceFile.h"
 
 #include <algorithm>
```
Comment (dlidstrom) [inline src/SourceLine.cpp:5]: Interesting, so <iterator> is included by <algorithm> by both gcc and clang. I think you can remove the #ifdef as this header really should be included on all platforms.

File: src/SourceLine.cpp [hunk 2]
```diff
@@ -3,11 +3,11 @@
...
 
 #include <algorithm>
+#ifdef WIN32
+#include <iterator>
+#endif
 
-SourceLine::SourceLine(const std::string& line, int lineNumber) {
-    m_line = line;
-    m_lineNumber = lineNumber;
-
```
Comment (dlidstrom) [inline src/SourceLine.cpp:10]: Can you put these on the line under? See SourceFile.cpp. Thanks.

**Review by dlidstrom — CHANGES_REQUESTED** [review]
Minor fixes needed.

**Comment by dlidstrom** [comment]
Hi @tyler97 any chance you can finish this?

**Comment by tyler97** [comment]
Hi @dlidstrom , yes I'll try to finish it this week, been busy moving and doing some work around the house so hadn't had a chance to finish it earlier. Thanks!

**Comment by DanielLidstromTicket** [comment]
Looks good to me, I can merge if you feel you're done.

**Comment by dlidstrom** [comment]
Looks good to me, I can merge if you feel you're done.

**Pull request merged**